src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/BeeWingListener.java

```diff
@@ -0,0 +1,54 @@
+package io.github.thebusybiscuit.slimefun4.implementation.listeners;
+
+import io.github.thebusybiscuit.slimefun4.utils.SlimefunUtils;
+import me.mrCookieSlime.Slimefun.Lists.SlimefunItems;
+import me.mrCookieSlime.Slimefun.SlimefunPlugin;
+import me.mrCookieSlime.Slimefun.api.Slimefun;
+import org.bukkit.Location;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.Listener;
+import org.bukkit.event.player.PlayerMoveEvent;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.potion.PotionEffect;
+import org.bukkit.potion.PotionEffectType;
+
+/**
+ * This {@link Listener} is responsible for the slow falling effect given to the player
+ * When Nearing the ground.
+ *
+ * @author beSnow
+ */
+public class BeeWingListener implements Listener {
+
+    public BeeWingListener(SlimefunPlugin plugin) {
+        plugin.getServer().getPluginManager().registerEvents(this, plugin);
+    }
+
+    @EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
+    public void onApproachGround(PlayerMoveEvent e) {
```

Comment:
This is ran on every little movement, it can run hundreds or thousands of times a second. Doing an item lookup EACH time is gonna REALLY hurt performance. You should do a ton of checks before ANYTHING to make sure this is exactly what you want!